src/ray/raylet/scheduling/cluster_resource_scheduler.h

```diff
@@ -413,6 +352,61 @@ class ClusterResourceScheduler {
 
   /// Return human-readable string for this scheduler state.
   std::string DebugString() const;
+
+ private:
+  struct Node {
+    Node(const NodeResources &resources)
+        : last_reported(resources), local_view(resources) {}
+
+    /// The resource information according to the last heartbeat reported by
+    /// this node.
+    /// NOTE(swang): For the local node, this field should be ignored because
+    /// we do not receive heartbeats from ourselves and the local view is
+    /// therefore always the most up-to-date.
+    NodeResources last_reported;
+    /// Our local view of the remote node's resources. This may be dirty
+    /// because it includes any resource requests that we allocated to this
+    /// node through spillback since our last heartbeat tick.
+    NodeResources local_view;
```

Comment:
Minor, but we could defensively make these private fields and add accessors that make it harder to mess up, e.g.,

ResetResources()  // updates both reported and local view
UpdateLocal()  // updates local only
GetLocal() const  // get local read view